storage/innobase/row/row0ins.cc

```diff
@@ -2773,6 +2768,30 @@ row_ins_clust_index_entry_low(
 
 			goto err_exit;
 		}
+	} else if (flags == (BTR_NO_UNDO_LOG_FLAG | BTR_NO_LOCKING_FLAG)
+		   && !index->table->has_spatial_index()
+		   && !index->table->versioned()
+		   && !index->table->is_temporary()
+		   && !index->table->n_rec_locks) {
+
+		ut_ad(index->table->skip_alter_undo);
+		ut_ad(!entry->is_metadata());
+		ut_ad(!index->table->is_active_ddl());
```

Comment:
It might be less confusing to rename `is_active_ddl()` to something like `is_native_online_ddl()`.